src/main/webapp/src/app/map/map.component.ts

```diff
@@ -1,77 +1,93 @@
 import { Component, AfterViewInit, ViewChild, ElementRef} from '@angular/core';
 import { LocationService } from '../location.service';
 import { Location } from 'src/app/models/location';
-import { HiddenGemService } from '../hidden-gem.service';
-
+import { AppComponent } from 'src/app/app.component';
 
 @Component({
   selector: 'app-map',
   templateUrl: `map.component.html`,
   styleUrls: ['./map.component.css']
 })
 
-export class MapComponent implements AfterViewInit {
-
+export class MapComponent {
+  
   @ViewChild("mapContainer", { static: false })
   mapContainer!: ElementRef;
   map!: google.maps.Map;
+  hiddenGems = this.appComponent.hiddenGems;
+  type!: string;
   location = {} as Location;
 
-  constructor(private locationService: LocationService) {
-      this.locationService.getLocation
+  constructor( private appComponent: AppComponent, private locationService: LocationService) {
+    this.locationService.getLocation
       .subscribe(location => {
         this.location = location;
-    })
+      })
   }
 
-  ngAfterViewInit(): void {
-    this.mapInitializer();
+  ngDoCheck() {
+    if (this.hiddenGems !== this.appComponent.hiddenGems) {
+      this.hiddenGems = this.appComponent.hiddenGems;
+      this.mapInitializer();
+    }
   }
-
+  
   mapInitializer(): void {
     // Coordinates fetched from the user's location to set the center of the map.
     let centerOfMapCoordinates = new google.maps.LatLng(this.location.lat, this.location.lng);
     let mapOptions: google.maps.MapOptions = {
       center: centerOfMapCoordinates,
-      zoom: 10
+      zoom: 15
     };
     this.map = new google.maps.Map(this.mapContainer.nativeElement, mapOptions);
+    var androidImg = '../../assets/images/android.png';
+    //Default Marker
```

Comment:
Nit - what is this default marker? Maybe you can remove this comment as I think userMarker is clear enough?